src/herder/test/TxSetTests.cpp

```diff
@@ -530,6 +570,68 @@ TEST_CASE("generalized tx set XDR conversion", "[txset]")
     }
 }
 
+TEST_CASE("generalized tx set with multiple txs per source account", "[txset]")
+{
+    VirtualClock clock;
+    auto cfg = getTestConfig();
+    cfg.LIMIT_TX_QUEUE_SOURCE_ACCOUNT = true;
+    cfg.LEDGER_PROTOCOL_VERSION =
+        static_cast<uint32_t>(GENERALIZED_TX_SET_PROTOCOL_VERSION);
+    cfg.TESTING_UPGRADE_LEDGER_PROTOCOL_VERSION =
+        static_cast<uint32_t>(GENERALIZED_TX_SET_PROTOCOL_VERSION);
+    Application::pointer app = createTestApplication(clock, cfg);
+    auto root = TestAccount::createRoot(*app);
+    int accountId = 1;
+
+    auto createTx = [&](int opCnt, int fee, bool unique) {
+        std::vector<Operation> ops;
+        for (int i = 0; i < opCnt; ++i)
+        {
+            ops.emplace_back(createAccount(
+                getAccount(std::to_string(accountId++)).getPublicKey(), 1));
+        }
+
+        if (unique)
+        {
+            // Create a new unique accounts to ensure there are no collisions
+            auto source =
+                root.create("unique " + std::to_string(accountId * -1),
```

Comment:
move `-` to "unique - "? This got me puzzled for a moment.

Reply:
oh I did this to avoid collisions with op generation above (it's also using accountId to get the destination address), but actually that should be fixed already with the "unique" prefix. Updated.